README.md

```diff
@@ -3,3 +3,33 @@
 This is the rust implementation of [OpenFeature](https://openfeature.dev), a vendor-agnostic abstraction library for evaluating feature flags.
 
 We support multiple data types for flags (numbers, strings, booleans, objects) as well as hooks, which can alter the lifecycle of a flag evaluation.
+
+
+## Installation
+
+```
+rust-sdk = { git = "https://github.com/open-feature/rust-sdk", branch = "main" }
+```
+
+## Usage
+
+### Initialization
+
+```rust
+use rust_sdk::Client;
+use rust_sdk::providers::NoopProvider;
+use rust_sdk::providers::traits::FeatureProvider;
+use rust_sdk::traits::ClientTraits;
+
+fn main() {
+    
+    let client = Client::<NoopProvider>::new(
+        "client-name".to_string(),
+        NoopProvider::new(),
+    );
```

Comment:
Looking at this, as well as the client, there doesn't seem to be a means of getting a client without specifying a provider, or a means of setting the global provider instance.

See [1.1.1, 1.1.2, and 1.1.3](https://github.com/open-feature/spec/blob/main/specification/sections/01-flag-evaluation.md#11-api-initialization-and-configuration).

Recently, I've [actually proposed an API similar](https://github.com/open-feature/ofep/issues/49) to what you show here, but it's not yet part of the spec.

Apologies if I'm misreading.

Reply:
You're right, you can't get the client without a provider. This was heavily influenced on how the go-sdk works ( interesting the use of a singleton and globals there... but that's another issue )

Reply:
> You're right, you can't get the client without a provider. This was heavily influenced on how the go-sdk works ( interesting the use of a singleton and globals there... but that's another issue )

I'm a bit confused by this comment :sweat:. The basic usage in Go looks something like this:

```go
	openfeature.SetProvider(openfeature.NoopProvider{})
	client := openfeature.NewClient("app")
	value, err := client.BooleanValue(
		context.Background(), "v2_enabled", false, openfeature.EvaluationContext{},
	)
```

It may be that internally, the client is created with a provider, but that's not the API the developer experiences.

Should we create a new issue for this?